Compilation issue with web server for LibreTiny on docker image

### The problem

captive_portal do not compile on LibreTiny platforms with 2025.7.0 and 2025.8.0-dev docker images

### Which version of ESPHome has the issue?

2025.7.0

### What type of installation are you using?

Docker

### What platform are you using?

BK72XX

### Component causing the issue

captive_portal

### YAML Config

```yaml
esphome:
  name: bk

bk72xx:
  board: wb3s

wifi:
  ssid: MySSID
  password: password1

captive_portal:
```

### Anything in the logs that might be useful for us?

```txt
INFO ESPHome 2025.7.0
INFO Reading configuration test2.yaml...
INFO Generating C++ source...
INFO Compiling app...
Processing bk (board: wb3s; framework: arduino; platform: libretiny @ 1.9.1)
----------------------------------------------------------------------------------------------------------------------
LibreTiny: Creating a virtual environment for Python dependencies
LibreTiny: Installing Python dependencies
Collecting ltchiptool<5.0,>=4.12.0
  Downloading ltchiptool-4.12.0-py3-none-any.whl.metadata (4.2 kB)
Collecting bitstruct<9.0.0,>=8.1.1 (from ltchiptool<5.0,>=4.12.0)
  Downloading bitstruct-8.21.0-cp312-cp312-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (7.3 kB)
Collecting bk7231tools<3.0.0,>=2.0.0 (from ltchiptool<5.0,>=4.12.0)
  Downloading bk7231tools-2.1.0-py3-none-any.whl.metadata (841 bytes)
Collecting click<9.0.0,>=8.1.3 (from ltchiptool<5.0,>=4.12.0)
  Downloading click-8.2.1-py3-none-any.whl.metadata (2.5 kB)
Collecting colorama<0.5.0,>=0.4.5 (from ltchiptool<5.0,>=4.12.0)
  Downloading colorama-0.4.6-py2.py3-none-any.whl.metadata (17 kB)
Collecting hexdump<4.0,>=3.3 (from ltchiptool<5.0,>=4.12.0)
  Downloading hexdump-3.3.zip (12 kB)
  Installing build dependencies: started
  Installing build dependencies: finished with status 'done'
  Getting requirements to build wheel: started
  Getting requirements to build wheel: finished with status 'done'
  Preparing metadata (pyproject.toml): started
  Preparing metadata (pyproject.toml): finished with status 'done'
Collecting importlib-metadata (from ltchiptool<5.0,>=4.12.0)
  Downloading importlib_metadata-8.7.0-py3-none-any.whl.metadata (4.8 kB)
Collecting prettytable<4.0.0,>=3.3.0 (from ltchiptool<5.0,>=4.12.0)
  Downloading prettytable-3.16.0-py3-none-any.whl.metadata (33 kB)
Collecting py-datastruct<2.0.0,>=1.0.0 (from ltchiptool<5.0,>=4.12.0)
  Downloading py_datastruct-1.1.0-py3-none-any.whl.metadata (10 kB)
Collecting pycryptodome<4.0.0,>=3.9.9 (from ltchiptool<5.0,>=4.12.0)
  Downloading pycryptodome-3.23.0-cp37-abi3-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (3.4 kB)
Collecting requests<3.0.0,>=2.31.0 (from ltchiptool<5.0,>=4.12.0)
  Downloading requests-2.32.4-py3-none-any.whl.metadata (4.9 kB)
Collecting semantic-version<3.0.0,>=2.10.0 (from ltchiptool<5.0,>=4.12.0)
  Downloading semantic_version-2.10.0-py2.py3-none-any.whl.metadata (9.7 kB)
Collecting xmodem<0.5.0,>=0.4.6 (from ltchiptool<5.0,>=4.12.0)
  Downloading xmodem-0.4.7-py3-none-any.whl.metadata (4.3 kB)
Collecting ymodem<2.0.0,>=1.5.1 (from ltchiptool<5.0,>=4.12.0)
  Downloading ymodem-1.5.1-py3-none-any.whl.metadata (5.6 kB)
Collecting pyserial<4.0,>=3.5 (from bk7231tools<3.0.0,>=2.0.0->ltchiptool<5.0,>=4.12.0)
  Downloading pyserial-3.5-py2.py3-none-any.whl.metadata (1.6 kB)
Collecting wcwidth (from prettytable<4.0.0,>=3.3.0->ltchiptool<5.0,>=4.12.0)
  Downloading wcwidth-0.2.13-py2.py3-none-any.whl.metadata (14 kB)
Collecting charset_normalizer<4,>=2 (from requests<3.0.0,>=2.31.0->ltchiptool<5.0,>=4.12.0)
  Downloading charset_normalizer-3.4.2-cp312-cp312-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (35 kB)
Collecting idna<4,>=2.5 (from requests<3.0.0,>=2.31.0->ltchiptool<5.0,>=4.12.0)
  Downloading idna-3.10-py3-none-any.whl.metadata (10 kB)
Collecting urllib3<3,>=1.21.1 (from requests<3.0.0,>=2.31.0->ltchiptool<5.0,>=4.12.0)
  Downloading urllib3-2.5.0-py3-none-any.whl.metadata (6.5 kB)
Collecting certifi>=2017.4.17 (from requests<3.0.0,>=2.31.0->ltchiptool<5.0,>=4.12.0)
  Downloading certifi-2025.7.14-py3-none-any.whl.metadata (2.4 kB)
Collecting ordered-set>=4.1.0 (from ymodem<2.0.0,>=1.5.1->ltchiptool<5.0,>=4.12.0)
  Downloading ordered_set-4.1.0-py3-none-any.whl.metadata (5.3 kB)
Collecting zipp>=3.20 (from importlib-metadata->ltchiptool<5.0,>=4.12.0)
  Downloading zipp-3.23.0-py3-none-any.whl.metadata (3.6 kB)
Downloading ltchiptool-4.12.0-py3-none-any.whl (296 kB)
Downloading bitstruct-8.21.0-cp312-cp312-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (83 kB)
Downloading bk7231tools-2.1.0-py3-none-any.whl (37 kB)
Downloading click-8.2.1-py3-none-any.whl (102 kB)
Downloading colorama-0.4.6-py2.py3-none-any.whl (25 kB)
Downloading prettytable-3.16.0-py3-none-any.whl (33 kB)
Downloading py_datastruct-1.1.0-py3-none-any.whl (28 kB)
Downloading pycryptodome-3.23.0-cp37-abi3-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (2.3 MB)
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 2.3/2.3 MB 62.7 MB/s eta 0:00:00
Downloading requests-2.32.4-py3-none-any.whl (64 kB)
Downloading semantic_version-2.10.0-py2.py3-none-any.whl (15 kB)
Downloading xmodem-0.4.7-py3-none-any.whl (35 kB)
Downloading ymodem-1.5.1-py3-none-any.whl (21 kB)
Downloading importlib_metadata-8.7.0-py3-none-any.whl (27 kB)
Downloading certifi-2025.7.14-py3-none-any.whl (162 kB)
Downloading charset_normalizer-3.4.2-cp312-cp312-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (148 kB)
Downloading idna-3.10-py3-none-any.whl (70 kB)
Downloading ordered_set-4.1.0-py3-none-any.whl (7.6 kB)
Downloading pyserial-3.5-py2.py3-none-any.whl (90 kB)
Downloading urllib3-2.5.0-py3-none-any.whl (129 kB)
Downloading zipp-3.23.0-py3-none-any.whl (10 kB)
Downloading wcwidth-0.2.13-py2.py3-none-any.whl (34 kB)
Building wheels for collected packages: hexdump
  Building wheel for hexdump (pyproject.toml): started
  Building wheel for hexdump (pyproject.toml): finished with status 'done'
  Created wheel for hexdump: filename=hexdump-3.3-py3-none-any.whl size=8932 sha256=01758aa00235e21104f689aafcd98c40d663b7cca8cd12af5b08114f7ff1a48f
  Stored in directory: /root/.cache/pip/wheels/5e/b9/b5/227a20e7e8bbdb2b17e46a087c0f0119059ee65fe8374cac18
Successfully built hexdump
Installing collected packages: xmodem, wcwidth, pyserial, hexdump, zipp, urllib3, semantic-version, pycryptodome, py-datastruct, prettytable, ordered-set, idna, colorama, click, charset_normalizer, certifi, bitstruct, ymodem, requests, importlib-metadata, bk7231tools, ltchiptool
Successfully installed bitstruct-8.21.0 bk7231tools-2.1.0 certifi-2025.7.14 charset_normalizer-3.4.2 click-8.2.1 colorama-0.4.6 hexdump-3.3 idna-3.10 importlib-metadata-8.7.0 ltchiptool-4.12.0 ordered-set-4.1.0 prettytable-3.16.0 py-datastruct-1.1.0 pycryptodome-3.23.0 pyserial-3.5 requests-2.32.4 semantic-version-2.10.0 urllib3-2.5.0 wcwidth-0.2.13 xmodem-0.4.7 ymodem-1.5.1 zipp-3.23.0
Warning! Non-Git installations are NOT SUPPORTED.
HARDWARE: BK7231T 120MHz, 256KB RAM, 1.03MB Flash
 - framework-arduino-api @ 2022.8.24+sha.237b10a
 - framework-beken-bdk @ 0.0.0+v2021.06.07.sha.6491b8c
 - library-flashdb @ 1.2.0+sha.d5c892f
 - library-freertos @ 9.0.0+sha.ff31b7d
 - library-freertos-port @ 2025.1.18+sha.dbfc408
 - library-lwip @ 2.2.0-bdk+sha.48da4a2
 - library-printf @ 6.1.0+sha.28a79bd
 - library-uf2ota @ 5.0.0+sha.f955412
PLATFORM VERSIONS:
 - libretiny @ 1.9.1
 - ltchiptool @ 4.12.0
CUSTOM OPTIONS:
 - fw_name = esphome
 - fw_version = 2025.7.0
Dependency Graph
|-- AsyncTCP @ 3.4.5
|-- ESPAsyncWebServer @ 3.7.10
|-- DNSServer @ 1.1.0
Compiling .pioenvs/bk/src/esphome/components/captive_portal/captive_portal.cpp.o
Compiling .pioenvs/bk/src/esphome/components/libretiny/core.cpp.o
Compiling .pioenvs/bk/src/esphome/components/libretiny/gpio_arduino.cpp.o
Compiling .pioenvs/bk/src/esphome/components/libretiny/lt_component.cpp.o
Compiling .pioenvs/bk/src/esphome/components/libretiny/preferences.cpp.o
Compiling .pioenvs/bk/src/esphome/components/md5/md5.cpp.o
Compiling .pioenvs/bk/src/esphome/components/mdns/mdns_component.cpp.o
Compiling .pioenvs/bk/src/esphome/components/mdns/mdns_libretiny.cpp.o
Compiling .pioenvs/bk/src/esphome/components/network/util.cpp.o
Compiling .pioenvs/bk/src/esphome/components/ota/ota_backend.cpp.o
Compiling .pioenvs/bk/src/esphome/components/ota/ota_backend_arduino_libretiny.cpp.o
Compiling .pioenvs/bk/src/esphome/components/safe_mode/safe_mode.cpp.o
Compiling .pioenvs/bk/src/esphome/components/web_server/ota/ota_web_server.cpp.o
Compiling .pioenvs/bk/src/esphome/components/web_server_base/web_server_base.cpp.o
Compiling .pioenvs/bk/src/esphome/components/wifi/wifi_component.cpp.o
Compiling .pioenvs/bk/src/esphome/components/wifi/wifi_component_libretiny.cpp.o
Compiling .pioenvs/bk/src/esphome/core/application.cpp.o
Compiling .pioenvs/bk/src/esphome/core/component.cpp.o
Compiling .pioenvs/bk/src/esphome/core/component_iterator.cpp.o
Compiling .pioenvs/bk/src/esphome/core/controller.cpp.o
Compiling .pioenvs/bk/src/esphome/core/entity_base.cpp.o
Compiling .pioenvs/bk/src/esphome/core/helpers.cpp.o
Compiling .pioenvs/bk/src/esphome/core/log.cpp.o
Compiling .pioenvs/bk/src/esphome/core/scheduler.cpp.o
Compiling .pioenvs/bk/src/esphome/core/string_ref.cpp.o
Compiling .pioenvs/bk/src/esphome/core/util.cpp.o
Compiling .pioenvs/bk/src/main.cpp.o
Compiling .pioenvs/bk/lib46e/RPAsyncTCP/SyncClient.cpp.o
In file included from /config/.esphome/platformio/platforms/libretiny/cores/common/base/config/lwipopts.h:12,
                 from /config/.esphome/platformio/platforms/libretiny/cores/beken-72xx/base/config/lwipopts.h:5,
                 from /config/.esphome/platformio/packages/library-lwip/src/include/lwip/opt.h:51,
                 from /config/.esphome/platformio/packages/library-lwip/src/include/lwip/ip_addr.h:40,
                 from /root/.platformio/lib/RPAsyncTCP/src/RPAsyncTCP.h:24,
                 from /root/.platformio/lib/RPAsyncTCP/src/SyncClient.cpp:7:
/config/.esphome/platformio/packages/library-lwip/port/lwipopts.h:418: warning: "TCP_MSS" redefined
  418 | #define TCP_MSS                 (1500 - 40)
      |
In file included from /root/.platformio/lib/RPAsyncTCP/src/SyncClient.h:14,
                 from /root/.platformio/lib/RPAsyncTCP/src/SyncClient.cpp:6:
/root/.platformio/lib/RPAsyncTCP/src/async_config.h:9: note: this is the location of the previous definition
    9 |   #define TCP_MSS       MBED_CONF_LWIP_TCP_MSS        //(1460)
      |
Archiving .pioenvs/bk/lib759/libESPAsyncWebServer.a
Archiving .pioenvs/bk/lib794/libDNSServer.a
Compiling .pioenvs/bk/core_bk72xx_arduino_libraries/WiFi/WiFiScan.cpp.o
Archiving .pioenvs/bk/libcore_bk72xx_arduino_fixups.a
Compiling .pioenvs/bk/core_bk72xx_arduino/main.cpp.o
Compiling .pioenvs/bk/core_bk72xx_arduino/wiring.c.o
Compiling .pioenvs/bk/core_bk72xx_arduino/wiring_analog.c.o
Compiling .pioenvs/bk/core_bk72xx_arduino/wiring_digital.c.o
Compiling .pioenvs/bk/core_bk72xx_arduino/wiring_irq.c.o
Compiling .pioenvs/bk/core_common_arduino_libraries/api/Serial/Serial.cpp.o
Compiling .pioenvs/bk/core_common_arduino_libraries/api/SoftwareSerial/SoftwareSerial.cpp.o
Compiling .pioenvs/bk/core_common_arduino_libraries/api/WiFi/WiFi.cpp.o
Compiling .pioenvs/bk/core_common_arduino_libraries/api/WiFi/WiFiAP.cpp.o
/root/.platformio/lib/RPAsyncTCP/src/async_config.h:9:25: error: 'MBED_CONF_LWIP_TCP_MSS' was not declared in this scope
    9 |   #define TCP_MSS       MBED_CONF_LWIP_TCP_MSS        //(1460)
      |                         ^~~~~~~~~~~~~~~~~~~~~~
/root/.platformio/lib/RPAsyncTCP/src/SyncClient.h:40:34: note: in expansion of macro 'TCP_MSS'
   40 |     SyncClient(size_t txBufLen = TCP_MSS);
      |                                  ^~~~~~~
/root/.platformio/lib/RPAsyncTCP/src/async_config.h:9:25: error: 'MBED_CONF_LWIP_TCP_MSS' was not declared in this scope
    9 |   #define TCP_MSS       MBED_CONF_LWIP_TCP_MSS        //(1460)
      |                         ^~~~~~~~~~~~~~~~~~~~~~
/root/.platformio/lib/RPAsyncTCP/src/SyncClient.h:41:55: note: in expansion of macro 'TCP_MSS'
   41 |     SyncClient(AsyncClient *client, size_t txBufLen = TCP_MSS);
      |                                                       ^~~~~~~
Compiling .pioenvs/bk/core_common_arduino_libraries/api/WiFi/WiFiEvents.cpp.o
Compiling .pioenvs/bk/core_common_arduino_libraries/api/WiFi/WiFiGeneric.cpp.o
Compiling .pioenvs/bk/core_common_arduino_libraries/api/WiFi/WiFiSTA.cpp.o
Compiling .pioenvs/bk/core_common_arduino_libraries/api/WiFi/WiFiScan.cpp.o
Compiling .pioenvs/bk/core_common_arduino_libraries/common/FS/FS.cpp.o
*** [.pioenvs/bk/lib46e/RPAsyncTCP/SyncClient.cpp.o] Error 1
============================================ [FAILED] Took 13.38 seconds ============================================
```

### Additional information

Tested for these HW platforms :
- bk72xx : fails
- rtl87xx : fails
- esp32 : success

Not tested on a Python native install but seems to work : @bdraco [#7230 [captive_portal] Broken on libretiny](https://github.com/esphome/issues/issues/7230)

tried the HA addon and it compiles there fine as well
```
INFO ESPHome 2025.7.0
INFO Reading configuration /config/esphome/issue9566.yaml...
INFO Generating C++ source...
INFO Compiling app...
Processing bk (board: wb3s; framework: arduino; platform: libretiny @ 1.9.1)
--------------------------------------------------------------------------------
Platform Manager: Installing libretiny @ 1.9.1
INFO Installing libretiny @ 1.9.1
Downloading  [####################################]  100%
Unpacking  [####################################]  100%
Platform Manager: libretiny@1.9.1 has been installed!
INFO libretiny@1.9.1 has been installed!
Tool Manager: Installing git+https://github.com/libretiny-eu/framework-beken-bdk#v2021.06.07
INFO Installing git+https://github.com/libretiny-eu/framework-beken-bdk#v2021.06.07
git version 2.39.5
Cloning into '/data/cache/platformio/cache/tmp/pkg-installing-e5w6g78_'...
Note: switching to '6491b8c001b737c790cba0bee16e548859d5e6fe'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by switching back to a branch.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -c with the switch command. Example:

  git switch -c <new-branch-name>

Or undo this operation with:

  git switch -

Turn off this advice by setting config variable advice.detachedHead to false

Tool Manager: framework-beken-bdk@0.0.0+v2021.06.07.sha.6491b8c has been installed!
INFO framework-beken-bdk@0.0.0+v2021.06.07.sha.6491b8c has been installed!
Tool Manager: Installing git+https://github.com/libretiny-eu/ArduinoCore-API#2022.08.24
INFO Installing git+https://github.com/libretiny-eu/ArduinoCore-API#2022.08.24
git version 2.39.5
Cloning into '/data/cache/platformio/cache/tmp/pkg-installing-9f34nmya'...
Note: switching to '237b10a4f6da14bf32e5e0963b7f28865216a60c'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by switching back to a branch.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -c with the switch command. Example:

  git switch -c <new-branch-name>

Or undo this operation with:

  git switch -

Turn off this advice by setting config variable advice.detachedHead to false

Tool Manager: framework-arduino-api@2022.8.24+sha.237b10a has been installed!
INFO framework-arduino-api@2022.8.24+sha.237b10a has been installed!
Tool Manager: Installing git+https://github.com/libretiny-eu/lwip#2.2.0-bdk
INFO Installing git+https://github.com/libretiny-eu/lwip#2.2.0-bdk
git version 2.39.5
Cloning into '/data/cache/platformio/cache/tmp/pkg-installing-4jcy0hls'...
Tool Manager: library-lwip@2.2.0-bdk+sha.48da4a2 has been installed!
INFO library-lwip@2.2.0-bdk+sha.48da4a2 has been installed!
Tool Manager: Installing git+https://github.com/libretiny-eu/library-freertos#9.0.0
INFO Installing git+https://github.com/libretiny-eu/library-freertos#9.0.0
git version 2.39.5
Cloning into '/data/cache/platformio/cache/tmp/pkg-installing-31e143in'...
Tool Manager: library-freertos@9.0.0+sha.ff31b7d has been installed!
INFO library-freertos@9.0.0+sha.ff31b7d has been installed!
Tool Manager: Installing git+https://github.com/libretiny-eu/library-freertos-port#2025.01.18
INFO Installing git+https://github.com/libretiny-eu/library-freertos-port#2025.01.18
git version 2.39.5
Cloning into '/data/cache/platformio/cache/tmp/pkg-installing-usib9aa_'...
Note: switching to 'dbfc408e47f57d3ca9a22e982faf0094c3cfc7a6'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by switching back to a branch.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -c with the switch command. Example:

  git switch -c <new-branch-name>

Or undo this operation with:

  git switch -

Turn off this advice by setting config variable advice.detachedHead to false

Tool Manager: library-freertos-port@2025.1.18+sha.dbfc408 has been installed!
INFO library-freertos-port@2025.1.18+sha.dbfc408 has been installed!
Tool Manager: Installing git+https://github.com/libretiny-eu/library-flashdb#1.2.0
INFO Installing git+https://github.com/libretiny-eu/library-flashdb#1.2.0
git version 2.39.5
Cloning into '/data/cache/platformio/cache/tmp/pkg-installing-lw_4wey1'...
Note: switching to 'd5c892f0c545884e9f0de0cf67215e426dc8109d'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by switching back to a branch.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -c with the switch command. Example:

  git switch -c <new-branch-name>

Or undo this operation with:

  git switch -

Turn off this advice by setting config variable advice.detachedHead to false

Tool Manager: library-flashdb@1.2.0+sha.d5c892f has been installed!
INFO library-flashdb@1.2.0+sha.d5c892f has been installed!
Tool Manager: Installing git+https://github.com/libretiny-eu/library-printf#6.1.0
INFO Installing git+https://github.com/libretiny-eu/library-printf#6.1.0
git version 2.39.5
Cloning into '/data/cache/platformio/cache/tmp/pkg-installing-_4k9fni4'...
Note: switching to '28a79bdd114da9c84880537dec2a1399d82823d3'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by switching back to a branch.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -c with the switch command. Example:

  git switch -c <new-branch-name>

Or undo this operation with:

  git switch -

Turn off this advice by setting config variable advice.detachedHead to false

Tool Manager: library-printf@6.1.0+sha.28a79bd has been installed!
INFO library-printf@6.1.0+sha.28a79bd has been installed!
Tool Manager: Installing git+https://github.com/libretiny-eu/library-uf2ota#5.0.0
INFO Installing git+https://github.com/libretiny-eu/library-uf2ota#5.0.0
git version 2.39.5
Cloning into '/data/cache/platformio/cache/tmp/pkg-installing-2w_syzfx'...
Note: switching to 'f955412ed1b196fb2b09f4470ed54dcd9b026d03'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by switching back to a branch.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -c with the switch command. Example:

  git switch -c <new-branch-name>

Or undo this operation with:

  git switch -

Turn off this advice by setting config variable advice.detachedHead to false

Tool Manager: library-uf2ota@5.0.0+sha.f955412 has been installed!
INFO library-uf2ota@5.0.0+sha.f955412 has been installed!
Tool Manager: Installing toolchain-gccarmnoneeabi @ ~1.100301.0
INFO Installing toolchain-gccarmnoneeabi @ ~1.100301.0
Downloading  [####################################]  100%          
Unpacking  [####################################]  100%          
Tool Manager: toolchain-gccarmnoneeabi@1.100301.220327 has been installed!
INFO toolchain-gccarmnoneeabi@1.100301.220327 has been installed!
Library Manager: Installing ESP32Async/AsyncTCP @ 3.4.5
INFO Installing ESP32Async/AsyncTCP @ 3.4.5
Downloading  [####################################]  100%
Unpacking  [####################################]  100%
Library Manager: AsyncTCP@3.4.5 has been installed!
INFO AsyncTCP@3.4.5 has been installed!
Library Manager: Installing ESP32Async/ESPAsyncWebServer @ 3.7.10
INFO Installing ESP32Async/ESPAsyncWebServer @ 3.7.10
Downloading  [####################################]  100%
Unpacking  [####################################]  100%
Library Manager: ESPAsyncWebServer@3.7.10 has been installed!
INFO ESPAsyncWebServer@3.7.10 has been installed!
Library Manager: Resolving dependencies...
INFO Resolving dependencies...
LibreTiny: Creating a virtual environment for Python dependencies
LibreTiny: Installing Python dependencies
Collecting ltchiptool<5.0,>=4.12.0
  Downloading ltchiptool-4.12.0-py3-none-any.whl.metadata (4.2 kB)
Collecting bitstruct<9.0.0,>=8.1.1 (from ltchiptool<5.0,>=4.12.0)
  Downloading bitstruct-8.21.0-cp312-cp312-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (7.3 kB)
Collecting bk7231tools<3.0.0,>=2.0.0 (from ltchiptool<5.0,>=4.12.0)
  Downloading bk7231tools-2.1.0-py3-none-any.whl.metadata (841 bytes)
Collecting click<9.0.0,>=8.1.3 (from ltchiptool<5.0,>=4.12.0)
  Downloading click-8.2.1-py3-none-any.whl.metadata (2.5 kB)
Collecting colorama<0.5.0,>=0.4.5 (from ltchiptool<5.0,>=4.12.0)
  Downloading colorama-0.4.6-py2.py3-none-any.whl.metadata (17 kB)
Collecting hexdump<4.0,>=3.3 (from ltchiptool<5.0,>=4.12.0)
  Downloading hexdump-3.3.zip (12 kB)
  Installing build dependencies: started
  Installing build dependencies: finished with status 'done'
  Getting requirements to build wheel: started
  Getting requirements to build wheel: finished with status 'done'
  Preparing metadata (pyproject.toml): started
  Preparing metadata (pyproject.toml): finished with status 'done'
Collecting importlib-metadata (from ltchiptool<5.0,>=4.12.0)
  Downloading importlib_metadata-8.7.0-py3-none-any.whl.metadata (4.8 kB)
Collecting prettytable<4.0.0,>=3.3.0 (from ltchiptool<5.0,>=4.12.0)
  Downloading prettytable-3.16.0-py3-none-any.whl.metadata (33 kB)
Collecting py-datastruct<2.0.0,>=1.0.0 (from ltchiptool<5.0,>=4.12.0)
  Downloading py_datastruct-1.1.0-py3-none-any.whl.metadata (10 kB)
Collecting pycryptodome<4.0.0,>=3.9.9 (from ltchiptool<5.0,>=4.12.0)
  Downloading pycryptodome-3.23.0-cp37-abi3-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (3.4 kB)
Collecting requests<3.0.0,>=2.31.0 (from ltchiptool<5.0,>=4.12.0)
  Downloading requests-2.32.4-py3-none-any.whl.metadata (4.9 kB)
Collecting semantic-version<3.0.0,>=2.10.0 (from ltchiptool<5.0,>=4.12.0)
  Downloading semantic_version-2.10.0-py2.py3-none-any.whl.metadata (9.7 kB)
Collecting xmodem<0.5.0,>=0.4.6 (from ltchiptool<5.0,>=4.12.0)
  Downloading xmodem-0.4.7-py3-none-any.whl.metadata (4.3 kB)
Collecting ymodem<2.0.0,>=1.5.1 (from ltchiptool<5.0,>=4.12.0)
  Downloading ymodem-1.5.1-py3-none-any.whl.metadata (5.6 kB)
Collecting pyserial<4.0,>=3.5 (from bk7231tools<3.0.0,>=2.0.0->ltchiptool<5.0,>=4.12.0)
  Downloading pyserial-3.5-py2.py3-none-any.whl.metadata (1.6 kB)
Collecting wcwidth (from prettytable<4.0.0,>=3.3.0->ltchiptool<5.0,>=4.12.0)
  Downloading wcwidth-0.2.13-py2.py3-none-any.whl.metadata (14 kB)
Collecting charset_normalizer<4,>=2 (from requests<3.0.0,>=2.31.0->ltchiptool<5.0,>=4.12.0)
  Downloading charset_normalizer-3.4.2-cp312-cp312-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (35 kB)
Collecting idna<4,>=2.5 (from requests<3.0.0,>=2.31.0->ltchiptool<5.0,>=4.12.0)
  Downloading idna-3.10-py3-none-any.whl.metadata (10 kB)
Collecting urllib3<3,>=1.21.1 (from requests<3.0.0,>=2.31.0->ltchiptool<5.0,>=4.12.0)
  Downloading urllib3-2.5.0-py3-none-any.whl.metadata (6.5 kB)
Collecting certifi>=2017.4.17 (from requests<3.0.0,>=2.31.0->ltchiptool<5.0,>=4.12.0)
  Downloading certifi-2025.7.14-py3-none-any.whl.metadata (2.4 kB)
Collecting ordered-set>=4.1.0 (from ymodem<2.0.0,>=1.5.1->ltchiptool<5.0,>=4.12.0)
  Downloading ordered_set-4.1.0-py3-none-any.whl.metadata (5.3 kB)
Collecting zipp>=3.20 (from importlib-metadata->ltchiptool<5.0,>=4.12.0)
  Downloading zipp-3.23.0-py3-none-any.whl.metadata (3.6 kB)
Downloading ltchiptool-4.12.0-py3-none-any.whl (296 kB)
Downloading bitstruct-8.21.0-cp312-cp312-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (83 kB)
Downloading bk7231tools-2.1.0-py3-none-any.whl (37 kB)
Downloading click-8.2.1-py3-none-any.whl (102 kB)
Downloading colorama-0.4.6-py2.py3-none-any.whl (25 kB)
Downloading prettytable-3.16.0-py3-none-any.whl (33 kB)
Downloading py_datastruct-1.1.0-py3-none-any.whl (28 kB)
Downloading pycryptodome-3.23.0-cp37-abi3-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (2.3 MB)
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 2.3/2.3 MB 14.6 MB/s eta 0:00:00
Downloading requests-2.32.4-py3-none-any.whl (64 kB)
Downloading semantic_version-2.10.0-py2.py3-none-any.whl (15 kB)
Downloading xmodem-0.4.7-py3-none-any.whl (35 kB)
Downloading ymodem-1.5.1-py3-none-any.whl (21 kB)
Downloading importlib_metadata-8.7.0-py3-none-any.whl (27 kB)
Downloading certifi-2025.7.14-py3-none-any.whl (162 kB)
Downloading charset_normalizer-3.4.2-cp312-cp312-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (148 kB)
Downloading idna-3.10-py3-none-any.whl (70 kB)
Downloading ordered_set-4.1.0-py3-none-any.whl (7.6 kB)
Downloading pyserial-3.5-py2.py3-none-any.whl (90 kB)
Downloading urllib3-2.5.0-py3-none-any.whl (129 kB)
Downloading zipp-3.23.0-py3-none-any.whl (10 kB)
Downloading wcwidth-0.2.13-py2.py3-none-any.whl (34 kB)
Building wheels for collected packages: hexdump
  Building wheel for hexdump (pyproject.toml): started
  Building wheel for hexdump (pyproject.toml): finished with status 'done'
  Created wheel for hexdump: filename=hexdump-3.3-py3-none-any.whl size=8932 sha256=d49718ec9574d7ee9f6c1bc7961205f68146442ba982023d58312b03b818bac9
  Stored in directory: /root/.cache/pip/wheels/5e/b9/b5/227a20e7e8bbdb2b17e46a087c0f0119059ee65fe8374cac18
Successfully built hexdump
Installing collected packages: xmodem, wcwidth, pyserial, hexdump, zipp, urllib3, semantic-version, pycryptodome, py-datastruct, prettytable, ordered-set, idna, colorama, click, charset_normalizer, certifi, bitstruct, ymodem, requests, importlib-metadata, bk7231tools, ltchiptool
Successfully installed bitstruct-8.21.0 bk7231tools-2.1.0 certifi-2025.7.14 charset_normalizer-3.4.2 click-8.2.1 colorama-0.4.6 hexdump-3.3 idna-3.10 importlib-metadata-8.7.0 ltchiptool-4.12.0 ordered-set-4.1.0 prettytable-3.16.0 py-datastruct-1.1.0 pycryptodome-3.23.0 pyserial-3.5 requests-2.32.4 semantic-version-2.10.0 urllib3-2.5.0 wcwidth-0.2.13 xmodem-0.4.7 ymodem-1.5.1 zipp-3.23.0
Warning! Non-Git installations are NOT SUPPORTED.
HARDWARE: BK7231T 120MHz, 256KB RAM, 1.03MB Flash
 - framework-arduino-api @ 2022.8.24+sha.237b10a 
 - framework-beken-bdk @ 0.0.0+v2021.06.07.sha.6491b8c 
 - library-flashdb @ 1.2.0+sha.d5c892f 
 - library-freertos @ 9.0.0+sha.ff31b7d 
 - library-freertos-port @ 2025.1.18+sha.dbfc408 
 - library-lwip @ 2.2.0-bdk+sha.48da4a2 
 - library-printf @ 6.1.0+sha.28a79bd 
 - library-uf2ota @ 5.0.0+sha.f955412 
PLATFORM VERSIONS:
 - libretiny @ 1.9.1
 - ltchiptool @ 4.12.0
CUSTOM OPTIONS:
 - fw_name = esphome
 - fw_version = 2025.7.0
Library Manager: Installing DNSServer
INFO:Library Manager:Installing DNSServer
Downloading 0% 10%
Unpacking 0% 10% 20% 30% 40% 50% 60% 70% 80% 90% 100%
Library Manager: DNSServer@1.1.0 has been installed!
INFO:Library Manager:DNSServer@1.1.0 has been installed!
Dependency Graph
|-- AsyncTCP @ 3.4.5
|-- ESPAsyncWebServer @ 3.7.10
|-- DNSServer @ 1.1.0
Compiling .pioenvs/bk/src/esphome/components/captive_portal/captive_portal.cpp.o
Compiling .pioenvs/bk/src/esphome/components/libretiny/core.cpp.o
Compiling .pioenvs/bk/src/esphome/components/libretiny/gpio_arduino.cpp.o
Compiling .pioenvs/bk/src/esphome/components/libretiny/helpers.cpp.o
Compiling .pioenvs/bk/src/esphome/components/libretiny/lt_component.cpp.o
Compiling .pioenvs/bk/src/esphome/components/libretiny/preferences.cpp.o
Compiling .pioenvs/bk/src/esphome/components/md5/md5.cpp.o
Compiling .pioenvs/bk/src/esphome/components/mdns/mdns_component.cpp.o
Compiling .pioenvs/bk/src/esphome/components/mdns/mdns_libretiny.cpp.o
Compiling .pioenvs/bk/src/esphome/components/network/util.cpp.o
Compiling .pioenvs/bk/src/esphome/components/ota/ota_backend.cpp.o
Compiling .pioenvs/bk/src/esphome/components/ota/ota_backend_arduino_libretiny.cpp.o
Compiling .pioenvs/bk/src/esphome/components/safe_mode/safe_mode.cpp.o
Compiling .pioenvs/bk/src/esphome/components/web_server/ota/ota_web_server.cpp.o
Compiling .pioenvs/bk/src/esphome/components/web_server_base/web_server_base.cpp.o
Compiling .pioenvs/bk/src/esphome/components/wifi/wifi_component.cpp.o
Compiling .pioenvs/bk/src/esphome/components/wifi/wifi_component_libretiny.cpp.o
Compiling .pioenvs/bk/src/esphome/core/application.cpp.o
Compiling .pioenvs/bk/src/esphome/core/color.cpp.o
Compiling .pioenvs/bk/src/esphome/core/component.cpp.o
Compiling .pioenvs/bk/src/esphome/core/component_iterator.cpp.o
Compiling .pioenvs/bk/src/esphome/core/controller.cpp.o
Compiling .pioenvs/bk/src/esphome/core/entity_base.cpp.o
Compiling .pioenvs/bk/src/esphome/core/helpers.cpp.o
Compiling .pioenvs/bk/src/esphome/core/log.cpp.o
Compiling .pioenvs/bk/src/esphome/core/scheduler.cpp.o
Compiling .pioenvs/bk/src/esphome/core/string_ref.cpp.o
Compiling .pioenvs/bk/src/esphome/core/time.cpp.o
Compiling .pioenvs/bk/src/esphome/core/util.cpp.o
Compiling .pioenvs/bk/src/main.cpp.o
Compiling .pioenvs/bk/lib219/AsyncTCP/AsyncTCP.cpp.o
Compiling .pioenvs/bk/lib09a/ESPAsyncWebServer/AsyncEventSource.cpp.o
Compiling .pioenvs/bk/lib09a/ESPAsyncWebServer/AsyncJson.cpp.o
Compiling .pioenvs/bk/lib09a/ESPAsyncWebServer/AsyncMessagePack.cpp.o
Compiling .pioenvs/bk/lib09a/ESPAsyncWebServer/AsyncWebHeader.cpp.o
Compiling .pioenvs/bk/lib09a/ESPAsyncWebServer/AsyncWebServerRequest.cpp.o
Compiling .pioenvs/bk/lib09a/ESPAsyncWebServer/AsyncWebSocket.cpp.o
Compiling .pioenvs/bk/lib09a/ESPAsyncWebServer/BackPort_SHA1Builder.cpp.o
Compiling .pioenvs/bk/lib09a/ESPAsyncWebServer/ChunkPrint.cpp.o
In file included from /data/cache/platformio/packages/framework-beken-bdk/beken378/common/include.h:6,
                 from /data/cache/platformio/platforms/libretiny/cores/beken-72xx/base/fixups/include.h:3,
                 from /data/cache/platformio/packages/framework-beken-bdk/beken378/os/include/rtos_pub.h:4,
                 from /data/cache/platformio/packages/framework-beken-bdk/beken378/func/mbedtls/mbedtls-port/inc/tls_rtos.h:6,
                 from /data/cache/platformio/packages/framework-beken-bdk/beken378/func/mbedtls/mbedtls-port/inc/tls_config.h:32,
                 from /data/cache/platformio/platforms/libretiny/cores/beken-72xx/base/config/tls_config.h:5,
                 from /data/cache/platformio/packages/framework-beken-bdk/beken378/func/mbedtls/mbedtls/include/mbedtls/sha1.h:29,
                 from .piolibdeps/bk/ESPAsyncWebServer/src/AsyncWebSocket.cpp:21:
/data/cache/platformio/packages/framework-beken-bdk/beken378/common/generic.h:154: warning: "htons" redefined
  154 | #define htons(x) __htons(x)
      | 
In file included from /data/cache/platformio/packages/library-lwip/src/include/lwip/ip6_addr.h:46,
                 from .piolibdeps/bk/AsyncTCP/src/AsyncTCP.h:21,
                 from .piolibdeps/bk/ESPAsyncWebServer/src/AsyncWebSocket.h:10,
                 from .piolibdeps/bk/ESPAsyncWebServer/src/AsyncWebSocket.cpp:4:
/data/cache/platformio/packages/library-lwip/src/include/lwip/def.h:119: note: this is the location of the previous definition
  119 | #define htons(x) lwip_htons(x)
      | 
In file included from /data/cache/platformio/packages/framework-beken-bdk/beken378/common/include.h:6,
                 from /data/cache/platformio/platforms/libretiny/cores/beken-72xx/base/fixups/include.h:3,
                 from /data/cache/platformio/packages/framework-beken-bdk/beken378/os/include/rtos_pub.h:4,
                 from /data/cache/platformio/packages/framework-beken-bdk/beken378/func/mbedtls/mbedtls-port/inc/tls_rtos.h:6,
                 from /data/cache/platformio/packages/framework-beken-bdk/beken378/func/mbedtls/mbedtls-port/inc/tls_config.h:32,
                 from /data/cache/platformio/platforms/libretiny/cores/beken-72xx/base/config/tls_config.h:5,
                 from /data/cache/platformio/packages/framework-beken-bdk/beken378/func/mbedtls/mbedtls/include/mbedtls/sha1.h:29,
                 from .piolibdeps/bk/ESPAsyncWebServer/src/AsyncWebSocket.cpp:21:
/data/cache/platformio/packages/framework-beken-bdk/beken378/common/generic.h:155: warning: "ntohs" redefined
  155 | #define ntohs(x) __ntohs(x)
      | 
In file included from /data/cache/platformio/packages/library-lwip/src/include/lwip/ip6_addr.h:46,
                 from .piolibdeps/bk/AsyncTCP/src/AsyncTCP.h:21,
                 from .piolibdeps/bk/ESPAsyncWebServer/src/AsyncWebSocket.h:10,
                 from .piolibdeps/bk/ESPAsyncWebServer/src/AsyncWebSocket.cpp:4:
/data/cache/platformio/packages/library-lwip/src/include/lwip/def.h:120: note: this is the location of the previous definition
  120 | #define ntohs(x) lwip_ntohs(x)
      | 
In file included from /data/cache/platformio/packages/framework-beken-bdk/beken378/common/include.h:6,
                 from /data/cache/platformio/platforms/libretiny/cores/beken-72xx/base/fixups/include.h:3,
                 from /data/cache/platformio/packages/framework-beken-bdk/beken378/os/include/rtos_pub.h:4,
                 from /data/cache/platformio/packages/framework-beken-bdk/beken378/func/mbedtls/mbedtls-port/inc/tls_rtos.h:6,
                 from /data/cache/platformio/packages/framework-beken-bdk/beken378/func/mbedtls/mbedtls-port/inc/tls_config.h:32,
                 from /data/cache/platformio/platforms/libretiny/cores/beken-72xx/base/config/tls_config.h:5,
                 from /data/cache/platformio/packages/framework-beken-bdk/beken378/func/mbedtls/mbedtls/include/mbedtls/sha1.h:29,
                 from .piolibdeps/bk/ESPAsyncWebServer/src/AsyncWebSocket.cpp:21:
/data/cache/platformio/packages/framework-beken-bdk/beken378/common/generic.h:156: warning: "htonl" redefined
  156 | #define htonl(x) __htonl(x)
      | 
In file included from /data/cache/platformio/packages/library-lwip/src/include/lwip/ip6_addr.h:46,
                 from .piolibdeps/bk/AsyncTCP/src/AsyncTCP.h:21,
                 from .piolibdeps/bk/ESPAsyncWebServer/src/AsyncWebSocket.h:10,
                 from .piolibdeps/bk/ESPAsyncWebServer/src/AsyncWebSocket.cpp:4:
/data/cache/platformio/packages/library-lwip/src/include/lwip/def.h:121: note: this is the location of the previous definition
  121 | #define htonl(x) lwip_htonl(x)
      | 
In file included from /data/cache/platformio/packages/framework-beken-bdk/beken378/common/include.h:6,
                 from /data/cache/platformio/platforms/libretiny/cores/beken-72xx/base/fixups/include.h:3,
                 from /data/cache/platformio/packages/framework-beken-bdk/beken378/os/include/rtos_pub.h:4,
                 from /data/cache/platformio/packages/framework-beken-bdk/beken378/func/mbedtls/mbedtls-port/inc/tls_rtos.h:6,
                 from /data/cache/platformio/packages/framework-beken-bdk/beken378/func/mbedtls/mbedtls-port/inc/tls_config.h:32,
                 from /data/cache/platformio/platforms/libretiny/cores/beken-72xx/base/config/tls_config.h:5,
                 from /data/cache/platformio/packages/framework-beken-bdk/beken378/func/mbedtls/mbedtls/include/mbedtls/sha1.h:29,
                 from .piolibdeps/bk/ESPAsyncWebServer/src/AsyncWebSocket.cpp:21:
/data/cache/platformio/packages/framework-beken-bdk/beken378/common/generic.h:157: warning: "ntohl" redefined
  157 | #define ntohl(x) __ntohl(x)
      | 
In file included from /data/cache/platformio/packages/library-lwip/src/include/lwip/ip6_addr.h:46,
                 from .piolibdeps/bk/AsyncTCP/src/AsyncTCP.h:21,
                 from .piolibdeps/bk/ESPAsyncWebServer/src/AsyncWebSocket.h:10,
                 from .piolibdeps/bk/ESPAsyncWebServer/src/AsyncWebSocket.cpp:4:
/data/cache/platformio/packages/library-lwip/src/include/lwip/def.h:122: note: this is the location of the previous definition
  122 | #define ntohl(x) lwip_ntohl(x)
      | 
Compiling .pioenvs/bk/lib09a/ESPAsyncWebServer/Middleware.cpp.o
Archiving .pioenvs/bk/lib219/libAsyncTCP.a
Compiling .pioenvs/bk/lib09a/ESPAsyncWebServer/WebAuthentication.cpp.o
Compiling .pioenvs/bk/lib09a/ESPAsyncWebServer/WebHandlers.cpp.o
Compiling .pioenvs/bk/lib09a/ESPAsyncWebServer/WebRequest.cpp.o
Compiling .pioenvs/bk/lib09a/ESPAsyncWebServer/WebResponses.cpp.o
Compiling .pioenvs/bk/lib09a/ESPAsyncWebServer/WebServer.cpp.o
Compiling .pioenvs/bk/libf85/DNSServer/DNSServer.cpp.o
Compiling .pioenvs/bk/board_wb3s/wb3s.c.o
Archiving .pioenvs/bk/libboard_wb3s.a
Compiling .pioenvs/bk/arduino_api/api/Common.cpp.o
Compiling .pioenvs/bk/arduino_api/api/IPAddress.cpp.o
Compiling .pioenvs/bk/arduino_api/api/PluggableUSB.cpp.o
Compiling .pioenvs/bk/arduino_api/api/Print.cpp.o
Compiling .pioenvs/bk/arduino_api/api/Stream.cpp.o
Compiling .pioenvs/bk/arduino_api/api/String.cpp.o
Compiling .pioenvs/bk/core_bk72xx_arduino_libraries/Serial/Serial.cpp.o
Compiling .pioenvs/bk/core_bk72xx_arduino_libraries/WiFi/WiFi.cpp.o
Archiving .pioenvs/bk/libarduino_api.a
Archiving .pioenvs/bk/libf85/libDNSServer.a
Compiling .pioenvs/bk/core_bk72xx_arduino_libraries/WiFi/WiFiAP.cpp.o
Compiling .pioenvs/bk/core_bk72xx_arduino_libraries/WiFi/WiFiEvents.cpp.o
Compiling .pioenvs/bk/core_bk72xx_arduino_libraries/WiFi/WiFiGeneric.cpp.o
Compiling .pioenvs/bk/core_bk72xx_arduino_libraries/WiFi/WiFiSTA.cpp.o
Compiling .pioenvs/bk/core_bk72xx_arduino_libraries/WiFi/WiFiScan.cpp.o
Archiving .pioenvs/bk/libcore_bk72xx_arduino_fixups.a
Compiling .pioenvs/bk/core_bk72xx_arduino/main.cpp.o
Archiving .pioenvs/bk/lib09a/libESPAsyncWebServer.a
Compiling .pioenvs/bk/core_bk72xx_arduino/wiring.c.o
Compiling .pioenvs/bk/core_bk72xx_arduino/wiring_analog.c.o
Compiling .pioenvs/bk/core_bk72xx_arduino/wiring_digital.c.o
Compiling .pioenvs/bk/core_bk72xx_arduino/wiring_irq.c.o
Compiling .pioenvs/bk/core_common_arduino_libraries/api/Serial/Serial.cpp.o
Compiling .pioenvs/bk/core_common_arduino_libraries/api/SoftwareSerial/SoftwareSerial.cpp.o
Compiling .pioenvs/bk/core_common_arduino_libraries/api/WiFi/WiFi.cpp.o
Compiling .pioenvs/bk/core_common_arduino_libraries/api/WiFi/WiFiAP.cpp.o
Compiling .pioenvs/bk/core_common_arduino_libraries/api/WiFi/WiFiEvents.cpp.o
Compiling .pioenvs/bk/core_common_arduino_libraries/api/WiFi/WiFiGeneric.cpp.o
Archiving .pioenvs/bk/libcore_bk72xx_arduino.a
Compiling .pioenvs/bk/core_common_arduino_libraries/api/WiFi/WiFiSTA.cpp.o
Compiling .pioenvs/bk/core_common_arduino_libraries/api/WiFi/WiFiScan.cpp.o
Archiving .pioenvs/bk/libcore_bk72xx_arduino_libraries.a
Compiling .pioenvs/bk/core_common_arduino_libraries/common/FS/FS.cpp.o
Compiling .pioenvs/bk/core_common_arduino_libraries/common/IPv6Address/IPv6Address.cpp.o
Compiling .pioenvs/bk/core_common_arduino_libraries/common/MD5/MD5MbedTLSImpl.cpp.o
Compiling .pioenvs/bk/core_common_arduino_libraries/common/Update/Update.cpp.o
Compiling .pioenvs/bk/core_common_arduino_libraries/common/Update/UpdateUtil.cpp.o
Compiling .pioenvs/bk/core_common_arduino_libraries/common/WiFiClient/LwIPClient.cpp.o
Compiling .pioenvs/bk/core_common_arduino_libraries/common/WiFiClient/LwIPRxBuffer.cpp.o
Compiling .pioenvs/bk/core_common_arduino_libraries/common/WiFiClient/MbedTLSClient.cpp.o
Compiling .pioenvs/bk/core_common_arduino_libraries/common/WiFiServer/LwIPServer.cpp.o
Compiling .pioenvs/bk/core_common_arduino_libraries/common/WiFiUdp/LwIPUdp.cpp.o
Compiling .pioenvs/bk/core_common_arduino_libraries/common/mDNS/LwIPmDNS.cpp.o
Compiling .pioenvs/bk/core_common_arduino_libraries/common/mDNS/mDNS.cpp.o
Compiling .pioenvs/bk/core_common_arduino_libraries/ext/HTTPClient/HTTPClient.cpp.o
Compiling .pioenvs/bk/core_common_arduino_libraries/ext/StreamString/StreamString.cpp.o
Compiling .pioenvs/bk/core_common_arduino_libraries/ext/WebServer/Parsing.cpp.o
Compiling .pioenvs/bk/core_common_arduino_libraries/ext/WebServer/WebServer.cpp.o
Compiling .pioenvs/bk/core_common_arduino_libraries/ext/WebServer/detail/mimetable.cpp.o
Compiling .pioenvs/bk/core_common_arduino_libraries/ext/WiFiMulti/WiFiMulti.cpp.o
Compiling .pioenvs/bk/core_common_arduino_libraries/ext/base64/base64.cpp.o
Compiling .pioenvs/bk/core_common_arduino_libraries/ext/base64/libb64/cdecode.c.o
Compiling .pioenvs/bk/core_common_arduino_libraries/ext/base64/libb64/cencode.c.o
Compiling .pioenvs/bk/core_common_arduino_libraries/ext/cbuf/cbuf.cpp.o
Compiling .pioenvs/bk/core_common_arduino_libraries/inline/Singletons.cpp.o
Archiving .pioenvs/bk/libcore_common_arduino_fixups.a
Compiling .pioenvs/bk/core_common_arduino/Events.cpp.o
Compiling .pioenvs/bk/core_common_arduino/common/abi.cpp.o
Compiling .pioenvs/bk/core_common_arduino/common/dtostrf.c.o
Compiling .pioenvs/bk/core_common_arduino/common/serial_event.cpp.o
Compiling .pioenvs/bk/core_common_arduino/main.c.o
Compiling .pioenvs/bk/core_common_arduino/posix/time.c.o
Compiling .pioenvs/bk/core_common_arduino/wiring/wiring.c.o
Compiling .pioenvs/bk/core_common_arduino/wiring/wiring_compat.cpp.o
Compiling .pioenvs/bk/core_common_arduino/wiring/wiring_custom.c.o
Compiling .pioenvs/bk/core_common_arduino/wiring/wiring_irq.c.o
Compiling .pioenvs/bk/core_common_arduino/wiring/wiring_math.cpp.o
Compiling .pioenvs/bk/core_common_arduino/wiring/wiring_private.c.o
Compiling .pioenvs/bk/core_common_arduino/wiring/wiring_shift.c.o
Compiling .pioenvs/bk/printf/src/printf/printf.c.o
Compiling .pioenvs/bk/flashdb/port/fal/src/fal.c.o
Compiling .pioenvs/bk/flashdb/port/fal/src/fal_flash.c.o
Compiling .pioenvs/bk/flashdb/port/fal/src/fal_partition.c.o
Compiling .pioenvs/bk/flashdb/port/fal/src/fal_rtt.c.o
Archiving .pioenvs/bk/libcore_common_arduino_libraries.a
Compiling .pioenvs/bk/flashdb/src/fdb.c.o
Compiling .pioenvs/bk/flashdb/src/fdb_file.c.o
Compiling .pioenvs/bk/flashdb/src/fdb_kvdb.c.o
Compiling .pioenvs/bk/flashdb/src/fdb_tsdb.c.o
Compiling .pioenvs/bk/flashdb/src/fdb_utils.c.o
Compiling .pioenvs/bk/uf2ota/src/uf2binpatch.c.o
Compiling .pioenvs/bk/uf2ota/src/uf2ota.c.o
Archiving .pioenvs/bk/libprintf.a
Compiling .pioenvs/bk/uf2ota/src/uf2priv.c.o
Archiving .pioenvs/bk/libcore_bk7231t_fixups.a
Archiving .pioenvs/bk/libcore_bk7231t.a
Compiling .pioenvs/bk/core_bk72xx_fixups/fixups/arch_main.c.o
Compiling .pioenvs/bk/core_bk72xx_fixups/fixups/ate_app.c.o
Compiling .pioenvs/bk/core_bk72xx_fixups/fixups/clock_cal.c.o
Archiving .pioenvs/bk/libflashdb.a
Archiving .pioenvs/bk/libuf2ota.a
Compiling .pioenvs/bk/core_bk72xx_fixups/fixups/clock_rtos.c.o
Compiling .pioenvs/bk/core_bk72xx_fixups/fixups/gcc10.c.o
Compiling .pioenvs/bk/core_bk72xx_fixups/fixups/intc.c.o
Compiling .pioenvs/bk/core_bk72xx_fixups/fixups/sys_ctrl.c.o
Compiling .pioenvs/bk/core_bk72xx/api/lt_cpu.c.o
Compiling .pioenvs/bk/core_bk72xx/api/lt_device.c.o
Compiling .pioenvs/bk/core_bk72xx/api/lt_flash.c.o
Compiling .pioenvs/bk/core_bk72xx/api/lt_init.c.o
Compiling .pioenvs/bk/core_bk72xx/api/lt_mem.c.o
Compiling .pioenvs/bk/core_bk72xx/api/lt_ota.c.o
Compiling .pioenvs/bk/core_bk72xx/api/lt_sleep.c.o
Compiling .pioenvs/bk/core_bk72xx/api/lt_wdt.c.o
Compiling .pioenvs/bk/core_bk72xx/port/fal_flash_bk72xx_port.c.o
Compiling .pioenvs/bk/core_bk72xx/port/printf.c.o
Compiling .pioenvs/bk/core_bk72xx/wraps/BkDriverFlash.c.o
Compiling .pioenvs/bk/core_bk72xx/wraps/wlan_ui.c.o
Archiving .pioenvs/bk/libcore_bk72xx_fixups.a
Compiling .pioenvs/bk/core_common_fixups/fixups/malloc.c.o
Archiving .pioenvs/bk/libcore_bk72xx.a
Compiling .pioenvs/bk/core_common/api/lt_cpu.c.o
Compiling .pioenvs/bk/core_common/api/lt_device.c.o
Compiling .pioenvs/bk/core_common/api/lt_flash.c.o
Archiving .pioenvs/bk/libcore_common_fixups.a
Compiling .pioenvs/bk/core_common/api/lt_mem.c.o
Compiling .pioenvs/bk/core_common/api/lt_ota.c.o
Archiving .pioenvs/bk/libcore_common_arduino.a
Compiling .pioenvs/bk/core_common/api/lt_sleep.c.o
Compiling .pioenvs/bk/core_common/api/lt_utils.c.o
Compiling .pioenvs/bk/core_common/api/lt_wdt.c.o
Compiling .pioenvs/bk/core_common/lt_logger.c.o
Compiling .pioenvs/bk/core_common/lt_main.c.o
Compiling .pioenvs/bk/core_common/posix/itoa.c.o
Compiling .pioenvs/bk/core_common/posix/strcasecmp.c.o
Compiling .pioenvs/bk/core_common/posix/strdup.c.o
Compiling .pioenvs/bk/core_common/posix/strptime.c.o
Compiling .pioenvs/bk/core_common/wraps/putchar.c.o
Compiling .pioenvs/bk/core_common/wraps/puts.c.o
Compiling .pioenvs/bk/bdk_ble_4_2/ble.c.o
Archiving .pioenvs/bk/libcore_common.a
Compiling .pioenvs/bk/bdk_ble_4_2/modules/app/src/app_ble.c.o
Compiling .pioenvs/bk/bdk_ble_4_2/modules/app/src/app_comm.c.o
Compiling .pioenvs/bk/bdk_ble_4_2/modules/app/src/app_sdp.c.o
Compiling .pioenvs/bk/bdk_ble_4_2/modules/app/src/app_sec.c.o
Compiling .pioenvs/bk/bdk_ble_4_2/modules/app/src/app_task.c.o
Compiling .pioenvs/bk/bdk_ble_4_2/plactform/driver/ble_icu/ble_icu.c.o
Compiling .pioenvs/bk/bdk_ble_4_2/plactform/driver/uart/ble_uart.c.o
Compiling .pioenvs/bk/bdk_ble_4_2/plactform/modules/arch/ble_arch_main.c.o
Compiling .pioenvs/bk/bdk_ble_4_2/plactform/modules/common/RomCallFlash.c.o
Compiling .pioenvs/bk/bdk_ble_4_2/plactform/modules/dbg/dbg.c.o
Compiling .pioenvs/bk/bdk_ble_4_2/plactform/modules/dbg/dbg_mwsgen.c.o
Compiling .pioenvs/bk/bdk_ble_4_2/plactform/modules/dbg/dbg_swdiag.c.o
Compiling .pioenvs/bk/bdk_ble_4_2/plactform/modules/dbg/dbg_task.c.o
Compiling .pioenvs/bk/bdk_ble_4_2/plactform/modules/rf/src/ble_rf_xvr.c.o
Compiling .pioenvs/bk/bdk_ble_4_2/profiles/comm/src/comm.c.o
Compiling .pioenvs/bk/bdk_ble_4_2/profiles/comm/src/comm_task.c.o
Compiling .pioenvs/bk/bdk_ble_4_2/profiles/prf/src/prf.c.o
Compiling .pioenvs/bk/bdk_ble_4_2/profiles/prf/src/prf_utils.c.o
Compiling .pioenvs/bk/bdk_ble_4_2/profiles/sdp/src/sdp_service.c.o
Compiling .pioenvs/bk/bdk_ble_4_2/profiles/sdp/src/sdp_service_task.c.o
Compiling .pioenvs/bk/bdk_driver_spi/spi.c.o
Compiling .pioenvs/bk/bdk_driver_spi/spi_master.c.o
Compiling .pioenvs/bk/bdk_driver_spi/spi_slave.c.o
Compiling .pioenvs/bk/bdk_mbedtls/mbedtls-port/src/timing_alt.c.o
Compiling .pioenvs/bk/bdk_mbedtls/mbedtls-port/src/tls_certificate.c.o
Compiling .pioenvs/bk/bdk_mbedtls/mbedtls-port/src/tls_client.c.o
Archiving .pioenvs/bk/libbdk_driver_spi.a
Archiving .pioenvs/bk/libbdk_ble_4_2.a
Compiling .pioenvs/bk/bdk_mbedtls/mbedtls-port/src/tls_hardware.c.o
Compiling .pioenvs/bk/bdk_mbedtls/mbedtls-port/src/tls_mem.c.o
Compiling .pioenvs/bk/bdk_mbedtls/mbedtls-port/src/tls_net.c.o
Compiling .pioenvs/bk/bdk_mbedtls/mbedtls/library/aes.c.o
Compiling .pioenvs/bk/bdk_mbedtls/mbedtls/library/aesni.c.o
Compiling .pioenvs/bk/bdk_mbedtls/mbedtls/library/arc4.c.o
Compiling .pioenvs/bk/bdk_mbedtls/mbedtls/library/asn1parse.c.o
Compiling .pioenvs/bk/bdk_mbedtls/mbedtls/library/asn1write.c.o
Compiling .pioenvs/bk/bdk_mbedtls/mbedtls/library/base64.c.o
Compiling .pioenvs/bk/bdk_mbedtls/mbedtls/library/bignum.c.o
Compiling .pioenvs/bk/bdk_mbedtls/mbedtls/library/blowfish.c.o
Compiling .pioenvs/bk/bdk_mbedtls/mbedtls/library/camellia.c.o
Compiling .pioenvs/bk/bdk_mbedtls/mbedtls/library/ccm.c.o
Compiling .pioenvs/bk/bdk_mbedtls/mbedtls/library/certs.c.o
Compiling .pioenvs/bk/bdk_mbedtls/mbedtls/library/cipher.c.o
Compiling .pioenvs/bk/bdk_mbedtls/mbedtls/library/cipher_wrap.c.o
Compiling .pioenvs/bk/bdk_mbedtls/mbedtls/library/cmac.c.o
Compiling .pioenvs/bk/bdk_mbedtls/mbedtls/library/ctr_drbg.c.o
Compiling .pioenvs/bk/bdk_mbedtls/mbedtls/library/debug.c.o
Compiling .pioenvs/bk/bdk_mbedtls/mbedtls/library/des.c.o
Compiling .pioenvs/bk/bdk_mbedtls/mbedtls/library/dhm.c.o
Compiling .pioenvs/bk/bdk_mbedtls/mbedtls/library/ecdh.c.o
Compiling .pioenvs/bk/bdk_mbedtls/mbedtls/library/ecdsa.c.o
Compiling .pioenvs/bk/bdk_mbedtls/mbedtls/library/ecjpake.c.o
Compiling .pioenvs/bk/bdk_mbedtls/mbedtls/library/ecp.c.o
Compiling .pioenvs/bk/bdk_mbedtls/mbedtls/library/ecp_curves.c.o
Compiling .pioenvs/bk/bdk_mbedtls/mbedtls/library/entropy.c.o
Compiling .pioenvs/bk/bdk_mbedtls/mbedtls/library/entropy_poll.c.o
Compiling .pioenvs/bk/bdk_mbedtls/mbedtls/library/error.c.o
Compiling .pioenvs/bk/bdk_mbedtls/mbedtls/library/gcm.c.o
Compiling .pioenvs/bk/bdk_mbedtls/mbedtls/library/havege.c.o
Compiling .pioenvs/bk/bdk_mbedtls/mbedtls/library/hmac_drbg.c.o
Compiling .pioenvs/bk/bdk_mbedtls/mbedtls/library/md.c.o
Compiling .pioenvs/bk/bdk_mbedtls/mbedtls/library/md2.c.o
Compiling .pioenvs/bk/bdk_mbedtls/mbedtls/library/md4.c.o
Compiling .pioenvs/bk/bdk_mbedtls/mbedtls/library/md5.c.o
Compiling .pioenvs/bk/bdk_mbedtls/mbedtls/library/md_wrap.c.o
Compiling .pioenvs/bk/bdk_mbedtls/mbedtls/library/memory_buffer_alloc.c.o
Compiling .pioenvs/bk/bdk_mbedtls/mbedtls/library/net_sockets.c.o
Compiling .pioenvs/bk/bdk_mbedtls/mbedtls/library/oid.c.o
Compiling .pioenvs/bk/bdk_mbedtls/mbedtls/library/padlock.c.o
Compiling .pioenvs/bk/bdk_mbedtls/mbedtls/library/pem.c.o
Compiling .pioenvs/bk/bdk_mbedtls/mbedtls/library/pk.c.o
Compiling .pioenvs/bk/bdk_mbedtls/mbedtls/library/pk_wrap.c.o
Compiling .pioenvs/bk/bdk_mbedtls/mbedtls/library/pkcs11.c.o
Compiling .pioenvs/bk/bdk_mbedtls/mbedtls/library/pkcs12.c.o
Compiling .pioenvs/bk/bdk_mbedtls/mbedtls/library/pkcs5.c.o
Compiling .pioenvs/bk/bdk_mbedtls/mbedtls/library/pkparse.c.o
Compiling .pioenvs/bk/bdk_mbedtls/mbedtls/library/pkwrite.c.o
Compiling .pioenvs/bk/bdk_mbedtls/mbedtls/library/platform.c.o
Compiling .pioenvs/bk/bdk_mbedtls/mbedtls/library/ripemd160.c.o
Compiling .pioenvs/bk/bdk_mbedtls/mbedtls/library/rsa.c.o
Compiling .pioenvs/bk/bdk_mbedtls/mbedtls/library/sha1.c.o
Compiling .pioenvs/bk/bdk_mbedtls/mbedtls/library/sha256.c.o
Compiling .pioenvs/bk/bdk_mbedtls/mbedtls/library/sha512.c.o
Compiling .pioenvs/bk/bdk_mbedtls/mbedtls/library/ssl_cache.c.o
Compiling .pioenvs/bk/bdk_mbedtls/mbedtls/library/ssl_ciphersuites.c.o
Compiling .pioenvs/bk/bdk_mbedtls/mbedtls/library/ssl_cli.c.o
Compiling .pioenvs/bk/bdk_mbedtls/mbedtls/library/ssl_cookie.c.o
Compiling .pioenvs/bk/bdk_mbedtls/mbedtls/library/ssl_srv.c.o
Compiling .pioenvs/bk/bdk_mbedtls/mbedtls/library/ssl_ticket.c.o
Compiling .pioenvs/bk/bdk_mbedtls/mbedtls/library/ssl_tls.c.o
Compiling .pioenvs/bk/bdk_mbedtls/mbedtls/library/threading.c.o
Compiling .pioenvs/bk/bdk_mbedtls/mbedtls/library/timing.c.o
Compiling .pioenvs/bk/bdk_mbedtls/mbedtls/library/version.c.o
Compiling .pioenvs/bk/bdk_mbedtls/mbedtls/library/version_features.c.o
Compiling .pioenvs/bk/bdk_mbedtls/mbedtls/library/x509.c.o
Compiling .pioenvs/bk/bdk_mbedtls/mbedtls/library/x509_create.c.o
Compiling .pioenvs/bk/bdk_mbedtls/mbedtls/library/x509_crl.c.o
Compiling .pioenvs/bk/bdk_mbedtls/mbedtls/library/x509_crt.c.o
Compiling .pioenvs/bk/bdk_mbedtls/mbedtls/library/x509_csr.c.o
Compiling .pioenvs/bk/bdk_mbedtls/mbedtls/library/x509write_crt.c.o
Compiling .pioenvs/bk/bdk_mbedtls/mbedtls/library/x509write_csr.c.o
Compiling .pioenvs/bk/bdk_mbedtls/mbedtls/library/xtea.c.o
Compiling .pioenvs/bk/lwip_bdk/port/ethernetif.c.o
Compiling .pioenvs/bk/lwip_bdk/port/net.c.o
Compiling .pioenvs/bk/lwip_bdk/port/sys_arch.c.o
Compiling .pioenvs/bk/lwip_bdk/src/api/api_lib.c.o
Compiling .pioenvs/bk/lwip_bdk/src/api/api_msg.c.o
Compiling .pioenvs/bk/lwip_bdk/src/api/err.c.o
Compiling .pioenvs/bk/lwip_bdk/src/api/if_api.c.o
Compiling .pioenvs/bk/lwip_bdk/src/api/netbuf.c.o
Compiling .pioenvs/bk/lwip_bdk/src/api/netdb.c.o
Compiling .pioenvs/bk/lwip_bdk/src/api/netifapi.c.o
Compiling .pioenvs/bk/lwip_bdk/src/api/sockets.c.o
Archiving .pioenvs/bk/libbdk_mbedtls.a
Compiling .pioenvs/bk/lwip_bdk/src/api/tcpip.c.o
Compiling .pioenvs/bk/lwip_bdk/src/apps/mdns/mdns.c.o
Compiling .pioenvs/bk/lwip_bdk/src/apps/mdns/mdns_domain.c.o
Compiling .pioenvs/bk/lwip_bdk/src/apps/mdns/mdns_out.c.o
Compiling .pioenvs/bk/lwip_bdk/src/apps/sntp/sntp.c.o
Compiling .pioenvs/bk/lwip_bdk/src/core/altcp.c.o
Compiling .pioenvs/bk/lwip_bdk/src/core/altcp_alloc.c.o
Compiling .pioenvs/bk/lwip_bdk/src/core/altcp_tcp.c.o
Compiling .pioenvs/bk/lwip_bdk/src/core/def.c.o
Compiling .pioenvs/bk/lwip_bdk/src/core/dns.c.o
Compiling .pioenvs/bk/lwip_bdk/src/core/inet_chksum.c.o
Compiling .pioenvs/bk/lwip_bdk/src/core/init.c.o
Compiling .pioenvs/bk/lwip_bdk/src/core/ip.c.o
Compiling .pioenvs/bk/lwip_bdk/src/core/ipv4/acd.c.o
Compiling .pioenvs/bk/lwip_bdk/src/core/ipv4/autoip.c.o
Compiling .pioenvs/bk/lwip_bdk/src/core/ipv4/dhcp.c.o
Compiling .pioenvs/bk/lwip_bdk/src/core/ipv4/etharp.c.o
Compiling .pioenvs/bk/lwip_bdk/src/core/ipv4/icmp.c.o
Compiling .pioenvs/bk/lwip_bdk/src/core/ipv4/igmp.c.o
Compiling .pioenvs/bk/lwip_bdk/src/core/ipv4/ip4.c.o
Compiling .pioenvs/bk/lwip_bdk/src/core/ipv4/ip4_addr.c.o
Compiling .pioenvs/bk/lwip_bdk/src/core/ipv4/ip4_frag.c.o
Compiling .pioenvs/bk/lwip_bdk/src/core/ipv6/dhcp6.c.o
Compiling .pioenvs/bk/lwip_bdk/src/core/ipv6/ethip6.c.o
Compiling .pioenvs/bk/lwip_bdk/src/core/ipv6/icmp6.c.o
Compiling .pioenvs/bk/lwip_bdk/src/core/ipv6/inet6.c.o
Compiling .pioenvs/bk/lwip_bdk/src/core/ipv6/ip6.c.o
Compiling .pioenvs/bk/lwip_bdk/src/core/ipv6/ip6_addr.c.o
Compiling .pioenvs/bk/lwip_bdk/src/core/ipv6/ip6_frag.c.o
Compiling .pioenvs/bk/lwip_bdk/src/core/ipv6/mld6.c.o
Compiling .pioenvs/bk/lwip_bdk/src/core/ipv6/nd6.c.o
Compiling .pioenvs/bk/lwip_bdk/src/core/mem.c.o
Compiling .pioenvs/bk/lwip_bdk/src/core/memp.c.o
Compiling .pioenvs/bk/lwip_bdk/src/core/netif.c.o
Compiling .pioenvs/bk/lwip_bdk/src/core/pbuf.c.o
Compiling .pioenvs/bk/lwip_bdk/src/core/raw.c.o
Compiling .pioenvs/bk/lwip_bdk/src/core/stats.c.o
Compiling .pioenvs/bk/lwip_bdk/src/core/sys.c.o
Compiling .pioenvs/bk/lwip_bdk/src/core/tcp.c.o
Compiling .pioenvs/bk/lwip_bdk/src/core/tcp_in.c.o
Compiling .pioenvs/bk/lwip_bdk/src/core/tcp_out.c.o
Compiling .pioenvs/bk/lwip_bdk/src/core/timeouts.c.o
Compiling .pioenvs/bk/lwip_bdk/src/core/udp.c.o
Compiling .pioenvs/bk/lwip_bdk/src/netif/ethernet.c.o
Compiling .pioenvs/bk/bdk_freertos_thumb/os/FreeRTOSv9.0.0/rtos_pub.c.o
Compiling .pioenvs/bk/bdk_freertos_thumb/os/mem_arch.c.o
Compiling .pioenvs/bk/bdk_freertos_thumb/os/str_arch.c.o
Compiling .pioenvs/bk/freertos_port/beken-bdk/heap_4.c.o
Compiling .pioenvs/bk/freertos_port/beken-bdk/port.c.o
Compiling .pioenvs/bk/freertos/FreeRTOS/Source/croutine.c.o
Compiling .pioenvs/bk/freertos/FreeRTOS/Source/event_groups.c.o
Compiling .pioenvs/bk/freertos/FreeRTOS/Source/list.c.o
Compiling .pioenvs/bk/freertos/FreeRTOS/Source/queue.c.o
Archiving .pioenvs/bk/libbdk_freertos_thumb.a
Compiling .pioenvs/bk/freertos/FreeRTOS/Source/tasks.c.o
Compiling .pioenvs/bk/freertos/FreeRTOS/Source/timers.c.o
Compiling .pioenvs/bk/bdk_func/airkiss/airkiss_main.c.o
Archiving .pioenvs/bk/liblwip_bdk.a
Compiling .pioenvs/bk/bdk_func/airkiss/airkiss_pingpong.c.o
Compiling .pioenvs/bk/bdk_func/airkiss/bk_airkiss.c.o
Compiling .pioenvs/bk/bdk_func/base64/base_64.c.o
Compiling .pioenvs/bk/bdk_func/ble_wifi_exchange/ble_wifi_port.c.o
Compiling .pioenvs/bk/bdk_func/camera_intf/camera_intf.c.o
Compiling .pioenvs/bk/bdk_func/func.c.o
Archiving .pioenvs/bk/libfreertos_port.a
Compiling .pioenvs/bk/bdk_func/hostapd_intf/hostapd_intf.c.o
Compiling .pioenvs/bk/bdk_func/joint_up/role_launch.c.o
Compiling .pioenvs/bk/bdk_func/lwip_intf/dhcpd/dhcp-server-main.c.o
Compiling .pioenvs/bk/bdk_func/lwip_intf/dhcpd/dhcp-server.c.o
Compiling .pioenvs/bk/bdk_func/misc/pseudo_random.c.o
Compiling .pioenvs/bk/bdk_func/misc/start_type.c.o
Compiling .pioenvs/bk/bdk_func/misc/target_util.c.o
Archiving .pioenvs/bk/libfreertos.a
Compiling .pioenvs/bk/bdk_func/net_param_intf/net_param.c.o
Compiling .pioenvs/bk/bdk_func/power_save/ap_idle.c.o
Compiling .pioenvs/bk/bdk_func/power_save/manual_ps.c.o
Compiling .pioenvs/bk/bdk_func/power_save/mcu_ps.c.o
Compiling .pioenvs/bk/bdk_func/power_save/power_save.c.o
Compiling .pioenvs/bk/bdk_func/rwnx_intf/rw_ieee80211.c.o
Compiling .pioenvs/bk/bdk_func/rwnx_intf/rw_msdu.c.o
Compiling .pioenvs/bk/bdk_func/rwnx_intf/rw_msg_rx.c.o
Compiling .pioenvs/bk/bdk_func/rwnx_intf/rw_msg_tx.c.o
Compiling .pioenvs/bk/bdk_func/saradc_intf/saradc_intf.c.o
Compiling .pioenvs/bk/bdk_func/security/security_func.c.o
Compiling .pioenvs/bk/bdk_func/sim_uart/gpio_uart.c.o
Compiling .pioenvs/bk/bdk_func/sim_uart/pwm_uart.c.o
Compiling .pioenvs/bk/bdk_func/spidma_intf/spidma_intf.c.o
Compiling .pioenvs/bk/bdk_func/temp_detect/temp_detect.c.o
Compiling .pioenvs/bk/bdk_func/usb_plug/usb_plug.c.o
Compiling .pioenvs/bk/bdk_func/user_driver/BkDriverFlash.c.o
Compiling .pioenvs/bk/bdk_func/user_driver/BkDriverGpio.c.o
Compiling .pioenvs/bk/bdk_func/user_driver/BkDriverPwm.c.o
Compiling .pioenvs/bk/bdk_func/user_driver/BkDriverRng.c.o
Compiling .pioenvs/bk/bdk_func/user_driver/BkDriverTimer.c.o
Compiling .pioenvs/bk/bdk_func/user_driver/BkDriverUart.c.o
Compiling .pioenvs/bk/bdk_func/user_driver/BkDriverWdg.c.o
Compiling .pioenvs/bk/bdk_func/utf8/conv_utf8.c.o
Compiling .pioenvs/bk/bdk_func/video_transfer/video_transfer.c.o
Compiling .pioenvs/bk/bdk_driver/calendar/calendar.c.o
Compiling .pioenvs/bk/bdk_driver/common/dd.c.o
Compiling .pioenvs/bk/bdk_driver/common/drv_model.c.o
Compiling .pioenvs/bk/bdk_driver/dma/dma.c.o
Compiling .pioenvs/bk/bdk_driver/driver.c.o
Compiling .pioenvs/bk/bdk_driver/fft/fft.c.o
Compiling .pioenvs/bk/bdk_driver/flash/flash.c.o
Compiling .pioenvs/bk/bdk_driver/general_dma/general_dma.c.o
Archiving .pioenvs/bk/libbdk_func.a
Compiling .pioenvs/bk/bdk_driver/gpio/gpio.c.o
Compiling .pioenvs/bk/bdk_driver/i2c/i2c1.c.o
Compiling .pioenvs/bk/bdk_driver/i2c/i2c2.c.o
Compiling .pioenvs/bk/bdk_driver/i2s/i2s.c.o
Compiling .pioenvs/bk/bdk_driver/icu/icu.c.o
Compiling .pioenvs/bk/bdk_driver/irda/irda.c.o
Compiling .pioenvs/bk/bdk_driver/jpeg/jpeg_encoder.c.o
Compiling .pioenvs/bk/bdk_driver/macphy_bypass/mac_phy_bypass.c.o
Compiling .pioenvs/bk/bdk_driver/phy/phy_trident.c.o
Compiling .pioenvs/bk/bdk_driver/pwm/bk_timer.c.o
Compiling .pioenvs/bk/bdk_driver/pwm/mcu_ps_timer.c.o
Compiling .pioenvs/bk/bdk_driver/pwm/pwm.c.o
Compiling .pioenvs/bk/bdk_driver/pwm/pwm_bk7231n.c.o
Compiling .pioenvs/bk/bdk_driver/pwm/pwm_mutex.c.o
Compiling .pioenvs/bk/bdk_driver/qspi/qspi.c.o
Compiling .pioenvs/bk/bdk_driver/rw_pub/rw_platf_pub.c.o
Compiling .pioenvs/bk/bdk_driver/saradc/saradc.c.o
Compiling .pioenvs/bk/bdk_driver/security/hal_aes.c.o
Compiling .pioenvs/bk/bdk_driver/security/hal_sha.c.o
Compiling .pioenvs/bk/bdk_driver/security/security.c.o
Compiling .pioenvs/bk/bdk_driver/spidma/spidma.c.o
Compiling .pioenvs/bk/bdk_driver/sys_ctrl/sys_ctrl.c.o
Compiling .pioenvs/bk/bdk_driver/uart/uart.c.o
Compiling .pioenvs/bk/bdk_driver/wdt/wdt.c.o
Compiling .pioenvs/bk/bdk_app/http/lite-log.c.o
Compiling .pioenvs/bk/bdk_app/http/utils_httpc.c.o
Compiling .pioenvs/bk/bdk_app/http/utils_net.c.o
Compiling .pioenvs/bk/bdk_app/http/utils_timer.c.o
Compiling .pioenvs/bk/bdk_app/net_work/video_demo_main.c.o
Compiling .pioenvs/bk/bdk_app/net_work/video_demo_softap.c.o
Compiling .pioenvs/bk/bdk_app/net_work/video_demo_station.c.o
Compiling .pioenvs/bk/bdk_app/standalone-ap/sa_ap.c.o
Compiling .pioenvs/bk/bdk_app/standalone-station/sa_station.c.o
Compiling .pioenvs/bk/bdk_app/video_work/video_buffer.c.o
Compiling .pioenvs/bk/bdk_app/video_work/video_transfer_tcp.c.o
Compiling .pioenvs/bk/bdk_app/video_work/video_transfer_udp.c.o
Compiling .pioenvs/bk/bdk_core/app/app.c.o
Compiling .pioenvs/bk/bdk_core/app/config/param_config.c.o
Compiling .pioenvs/bk/bdk_core/driver/entry/boot_vectors.S.o
Compiling .pioenvs/bk/bdk_core/func/wlan_ui/wlan_ui.c.o
Compiling .pioenvs/bk/bdk_boot/boot_handlers_bk7231u.S.o
Archiving .pioenvs/bk/libbdk_boot.a
Archiving .pioenvs/bk/libbdk_driver.a
Archiving .pioenvs/bk/libbdk_app.a
Archiving .pioenvs/bk/libbdk_core.a
Linking .pioenvs/bk/raw_firmware.elf
|-- Image 1: raw_firmware.elf
|   |-- image_bk7231t_app.0x011000.rbl
|   |   |-- raw_firmware.bin
|   |-- image_bk7231t_app.0x011000.crc
|   |-- image_bk7231t_app.0x129F0A.rblh
|   |-- image_bk7231t_app.ota.rbl
|   |-- image_bk7231t_app.ota.ug.bin
RAM:   [====      ]  44.3% (used 116044 bytes from 262144 bytes)
Flash: [=====     ]  50.7% (used 549601 bytes from 1083136 bytes)
Building UF2 OTA image
|-- esphome_2025.7.0_wb3s_bk7231t_lt1.9.1.uf2
|-- firmware.uf2
|-- firmware.bin
========================= [SUCCESS] Took 69.49 seconds =========================
Warning! Non-Git installations are NOT SUPPORTED.
INFO Successfully compiled program.
ERROR Cannot upload Over the Air as the ota configuration is not present or does not include platform: esphome
```

okay it seems related to my docker install.
I've tried in HA addon and build is successful.

I've check again other targets on HA addon and I still have an issue compiling for rtl87xx but the errors are different :
```
Compiling .pioenvs/test/src/main.cpp.o
In file included from src/esphome/core/event_pool.h:8,
                 from src/esphome.h:39,
                 from src/main.cpp:3:
src/esphome/core/lock_free_queue.h: In member function 'bool esphome::NotifyingLockFreeQueue<T, SIZE>::push(T*)':
src/esphome/core/lock_free_queue.h:133:7: error: there are no arguments to 'xTaskNotifyGive' that depend on a template parameter, so a declaration of 'xTaskNotifyGive' must be available [-fpermissive]
  133 |       xTaskNotifyGive(task_to_notify_);
      |       ^~~~~~~~~~~~~~~
src/esphome/core/lock_free_queue.h:133:7: note: (if you use '-fpermissive', G++ will accept your code, but allowing the use of an undeclared name is deprecated)
Compiling .pioenvs/test/lib562/AsyncTCP/AsyncTCP.cpp.o
*** [.pioenvs/test/src/main.cpp.o] Error 1
.piolibdeps/test/AsyncTCP/src/AsyncTCP.cpp: In function 'void _send_async_event(lwip_tcp_event_packet_t*)':
.piolibdeps/test/AsyncTCP/src/AsyncTCP.cpp:202:3: error: 'xTaskNotifyGive' was not declared in this scope
  202 |   xTaskNotifyGive(_async_service_task_handle);
      |   ^~~~~~~~~~~~~~~
.piolibdeps/test/AsyncTCP/src/AsyncTCP.cpp: In function 'void _prepend_async_event(lwip_tcp_event_packet_t*)':
.piolibdeps/test/AsyncTCP/src/AsyncTCP.cpp:210:3: error: 'xTaskNotifyGive' was not declared in this scope
  210 |   xTaskNotifyGive(_async_service_task_handle);
      |   ^~~~~~~~~~~~~~~
.piolibdeps/test/AsyncTCP/src/AsyncTCP.cpp: In function 'void _async_service_task(void*)':
.piolibdeps/test/AsyncTCP/src/AsyncTCP.cpp:322:5: error: 'ulTaskNotifyTake' was not declared in this scope
  322 |     ulTaskNotifyTake(pdTRUE, ASYNC_TCP_MAX_TASK_SLEEP);
      |     ^~~~~~~~~~~~~~~~
*** [.pioenvs/test/lib562/AsyncTCP/AsyncTCP.cpp.o] Error 1
```

I've tried with 2025.7.5, but I am still getting the same error.

Tested with 2 different devices running generic-bk7231n-qfn32-tuya

````
Compiling .pioenvs/luz-cozinha/src/main.cpp.o
Compiling .pioenvs/luz-cozinha/liba63/AsyncTCP/AsyncTCP.cpp.o
Compiling .pioenvs/luz-cozinha/lib46e/RPAsyncTCP/AsyncPrinter.cpp.o
Compiling .pioenvs/luz-cozinha/lib46e/RPAsyncTCP/RPAsyncTCP.cpp.o
In file included from /config/.esphome/platformio/platforms/libretiny/cores/common/base/config/lwipopts.h:12,
                 from /config/.esphome/platformio/platforms/libretiny/cores/beken-72xx/base/config/lwipopts.h:5,
                 from /config/.esphome/platformio/packages/library-lwip/src/include/lwip/opt.h:51,
                 from /config/.esphome/platformio/packages/library-lwip/src/include/lwip/ip_addr.h:40,
                 from /root/.platformio/lib/RPAsyncTCP/src/RPAsyncTCP.h:24,
                 from /root/.platformio/lib/RPAsyncTCP/src/AsyncPrinter.h:26,
                 from /root/.platformio/lib/RPAsyncTCP/src/AsyncPrinter.cpp:26:
/config/.esphome/platformio/packages/library-lwip/port/lwipopts.h:410: warning: "TCP_MSS" redefined
  410 | #define TCP_MSS                 (1500 - 40)
      | 
In file included from /root/.platformio/lib/RPAsyncTCP/src/RPAsyncTCP.h:17,
                 from /root/.platformio/lib/RPAsyncTCP/src/AsyncPrinter.h:26,
                 from /root/.platformio/lib/RPAsyncTCP/src/AsyncPrinter.cpp:26:
/root/.platformio/lib/RPAsyncTCP/src/async_config.h:9: note: this is the location of the previous definition
    9 |   #define TCP_MSS       MBED_CONF_LWIP_TCP_MSS        //(1460)
      | 
In file included from /config/.esphome/platformio/platforms/libretiny/cores/common/base/config/lwipopts.h:12,
                 from /config/.esphome/platformio/platforms/libretiny/cores/beken-72xx/base/config/lwipopts.h:5,
                 from /config/.esphome/platformio/packages/library-lwip/src/include/lwip/opt.h:51,
                 from /config/.esphome/platformio/packages/library-lwip/src/include/lwip/ip_addr.h:40,
                 from /root/.platformio/lib/RPAsyncTCP/src/RPAsyncTCP.h:24,
                 from /root/.platformio/lib/RPAsyncTCP/src/RPAsyncTCP.cpp:7:
/config/.esphome/platformio/packages/library-lwip/port/lwipopts.h:410: warning: "TCP_MSS" redefined
  410 | #define TCP_MSS                 (1500 - 40)
      | 
In file included from /root/.platformio/lib/RPAsyncTCP/src/RPAsyncTCP.h:17,
                 from /root/.platformio/lib/RPAsyncTCP/src/RPAsyncTCP.cpp:7:
/root/.platformio/lib/RPAsyncTCP/src/async_config.h:9: note: this is the location of the previous definition
    9 |   #define TCP_MSS       MBED_CONF_LWIP_TCP_MSS        //(1460)
      | 
Archiving .pioenvs/luz-cozinha/liba63/libAsyncTCP.a
Compiling .pioenvs/luz-cozinha/lib46e/RPAsyncTCP/RPAsyncTCP_buffer.cpp.o
In file included from /config/.esphome/platformio/platforms/libretiny/cores/common/base/config/lwipopts.h:12,
                 from /config/.esphome/platformio/platforms/libretiny/cores/beken-72xx/base/config/lwipopts.h:5,
                 from /config/.esphome/platformio/packages/library-lwip/src/include/lwip/opt.h:51,
                 from /config/.esphome/platformio/packages/library-lwip/src/include/lwip/ip_addr.h:40,
                 from /root/.platformio/lib/RPAsyncTCP/src/RPAsyncTCP.h:24,
                 from /root/.platformio/lib/RPAsyncTCP/src/RPAsyncTCP_buffer.h:11,
                 from /root/.platformio/lib/RPAsyncTCP/src/RPAsyncTCP_buffer.cpp:8:
/config/.esphome/platformio/packages/library-lwip/port/lwipopts.h:410: warning: "TCP_MSS" redefined
  410 | #define TCP_MSS                 (1500 - 40)
      | 
In file included from /root/.platformio/lib/RPAsyncTCP/src/RPAsyncTCP.h:17,
                 from /root/.platformio/lib/RPAsyncTCP/src/RPAsyncTCP_buffer.h:11,
                 from /root/.platformio/lib/RPAsyncTCP/src/RPAsyncTCP_buffer.cpp:8:
/root/.platformio/lib/RPAsyncTCP/src/async_config.h:9: note: this is the location of the previous definition
    9 |   #define TCP_MSS       MBED_CONF_LWIP_TCP_MSS        //(1460)
      | 
Compiling .pioenvs/luz-cozinha/lib46e/RPAsyncTCP/SyncClient.cpp.o
In file included from /config/.esphome/platformio/platforms/libretiny/cores/common/base/config/lwipopts.h:12,
                 from /config/.esphome/platformio/platforms/libretiny/cores/beken-72xx/base/config/lwipopts.h:5,
                 from /config/.esphome/platformio/packages/library-lwip/src/include/lwip/opt.h:51,
                 from /config/.esphome/platformio/packages/library-lwip/src/include/lwip/ip_addr.h:40,
                 from /root/.platformio/lib/RPAsyncTCP/src/RPAsyncTCP.h:24,
                 from /root/.platformio/lib/RPAsyncTCP/src/SyncClient.cpp:7:
/config/.esphome/platformio/packages/library-lwip/port/lwipopts.h:410: warning: "TCP_MSS" redefined
  410 | #define TCP_MSS                 (1500 - 40)
      | 
In file included from /root/.platformio/lib/RPAsyncTCP/src/SyncClient.h:14,
                 from /root/.platformio/lib/RPAsyncTCP/src/SyncClient.cpp:6:
/root/.platformio/lib/RPAsyncTCP/src/async_config.h:9: note: this is the location of the previous definition
    9 |   #define TCP_MSS       MBED_CONF_LWIP_TCP_MSS        //(1460)
      | 
Compiling .pioenvs/luz-cozinha/lib46e/RPAsyncTCP/cbuf.cpp.o
Compiling .pioenvs/luz-cozinha/lib46e/RPAsyncTCP/debug.cpp.o
Compiling .pioenvs/luz-cozinha/lib46e/RPAsyncTCP/tcp_axtls.c.o
Compiling .pioenvs/luz-cozinha/libcf8/ESPAsyncWebServer/AsyncEventSource.cpp.o
/root/.platformio/lib/RPAsyncTCP/src/async_config.h:9:25: error: 'MBED_CONF_LWIP_TCP_MSS' was not declared in this scope
    9 |   #define TCP_MSS       MBED_CONF_LWIP_TCP_MSS        //(1460)
      |                         ^~~~~~~~~~~~~~~~~~~~~~
/root/.platformio/lib/RPAsyncTCP/src/SyncClient.h:40:34: note: in expansion of macro 'TCP_MSS'
   40 |     SyncClient(size_t txBufLen = TCP_MSS);
      |                                  ^~~~~~~
/root/.platformio/lib/RPAsyncTCP/src/async_config.h:9:25: error: 'MBED_CONF_LWIP_TCP_MSS' was not declared in this scope
    9 |   #define TCP_MSS       MBED_CONF_LWIP_TCP_MSS        //(1460)
      |                         ^~~~~~~~~~~~~~~~~~~~~~
/root/.platformio/lib/RPAsyncTCP/src/SyncClient.h:41:55: note: in expansion of macro 'TCP_MSS'
   41 |     SyncClient(AsyncClient *client, size_t txBufLen = TCP_MSS);
      |                                                       ^~~~~~~
Compiling .pioenvs/luz-cozinha/libcf8/ESPAsyncWebServer/AsyncJson.cpp.o
Compiling .pioenvs/luz-cozinha/libcf8/ESPAsyncWebServer/AsyncMessagePack.cpp.o
Compiling .pioenvs/luz-cozinha/libcf8/ESPAsyncWebServer/AsyncWebHeader.cpp.o
*** [.pioenvs/luz-cozinha/lib46e/RPAsyncTCP/SyncClient.cpp.o] Error 1
========================= [FAILED] Took 137.51 seconds =========================
````

Installed versions:
platform: libretiny @ 1.9.1
ESP32Async/AsyncTCP @ 3.4.5
ESP32Async/ESPAsyncWebServer @ 3.7.10

> > A quick work around that worked for me.
> > platformio_options: lib_ignore: - RPAsyncTCP
> 
> This might be obvious to some, but for me it wasn't. Cleaning the build files is seemlingly required for this to work.

Can you explain to me what to do because I keep getting a (slightly different) error even when ignoring the library?

```
HARDWARE: RTL8710BN 125MHz, 256KB RAM, 788KB Flash
 - framework-arduino-api @ 2022.8.24+sha.237b10a 
 - framework-realtek-amb1 @ 0.0.0+v2022.06.21.sha.c4e44ef 
 - library-flashdb @ 1.2.0+sha.d5c892f 
 - library-freertos @ 8.1.2+sha.776ae6c 
 - library-freertos-port @ 2025.1.18+sha.dbfc408 
 - library-lwip @ 2.1.3-amb1+sha.6297b80 
 - library-printf @ 6.1.0+sha.28a79bd 
 - library-uf2ota @ 5.0.0+sha.f955412 
PLATFORM VERSIONS:
 - libretiny @ 1.9.1
 - ltchiptool @ 4.12.1
CUSTOM OPTIONS:
 - fw_name = esphome
 - fw_version = 2025.8.2
Dependency Graph
|-- AsyncTCP @ 3.4.5
|-- ESPAsyncWebServer @ 3.7.10
|-- DNSServer @ 1.1.0
|-- noise-c @ 0.1.10
|-- ArduinoJson @ 7.4.2
Compiling .pioenvs/presa-sp22-a-1/lib9ee/AsyncTCP/AsyncTCP.cpp.o
Compiling .pioenvs/presa-sp22-a-1/libb7a/ESPAsyncWebServer/WebResponses.cpp.o
Compiling .pioenvs/presa-sp22-a-1/libb7a/ESPAsyncWebServer/WebServer.cpp.o
Compiling .pioenvs/presa-sp22-a-1/lib74a/DNSServer/DNSServer.cpp.o
.piolibdeps/presa-sp22-a-1/AsyncTCP/src/AsyncTCP.cpp: In function 'void _send_async_event(lwip_tcp_event_packet_t*)':
.piolibdeps/presa-sp22-a-1/AsyncTCP/src/AsyncTCP.cpp:202:3: error: 'xTaskNotifyGive' was not declared in this scope
  202 |   xTaskNotifyGive(_async_service_task_handle);
      |   ^~~~~~~~~~~~~~~
.piolibdeps/presa-sp22-a-1/AsyncTCP/src/AsyncTCP.cpp: In function 'void _prepend_async_event(lwip_tcp_event_packet_t*)':
.piolibdeps/presa-sp22-a-1/AsyncTCP/src/AsyncTCP.cpp:210:3: error: 'xTaskNotifyGive' was not declared in this scope
  210 |   xTaskNotifyGive(_async_service_task_handle);
      |   ^~~~~~~~~~~~~~~
.piolibdeps/presa-sp22-a-1/AsyncTCP/src/AsyncTCP.cpp: In function 'void _async_service_task(void*)':
.piolibdeps/presa-sp22-a-1/AsyncTCP/src/AsyncTCP.cpp:322:5: error: 'ulTaskNotifyTake' was not declared in this scope
  322 |     ulTaskNotifyTake(pdTRUE, ASYNC_TCP_MAX_TASK_SLEEP);
      |     ^~~~~~~~~~~~~~~~
*** [.pioenvs/presa-sp22-a-1/lib9ee/AsyncTCP/AsyncTCP.cpp.o] Error 1
```

--- a/esphome/components/heatpumpir/climate.py
+++ b/esphome/components/heatpumpir/climate.py
@@ -128,4 +128,4 @@ async def to_code(config):
 
     cg.add_library("tonia/HeatpumpIR", "1.0.37")
     if CORE.is_libretiny or CORE.is_esp32:
-        CORE.add_platformio_option("lib_ignore", "IRremoteESP8266")
+        CORE.add_platformio_option("lib_ignore", ["IRremoteESP8266"])

--- a/esphome/components/web_server_base/__init__.py
+++ b/esphome/components/web_server_base/__init__.py
@@ -40,5 +40,7 @@ async def to_code(config):
             cg.add_library("Update", None)
         if CORE.is_esp8266:
             cg.add_library("ESP8266WiFi", None)
+        if CORE.is_libretiny:
+            CORE.add_platformio_option("lib_ignore", ["ESPAsyncTCP", "RPAsyncTCP"])
         # https://github.com/ESP32Async/ESPAsyncWebServer/blob/main/library.json
         cg.add_library("ESP32Async/ESPAsyncWebServer", "3.7.10")

--- a/esphome/core/config.py
+++ b/esphome/core/config.py
@@ -396,7 +396,7 @@ async def add_includes(includes: list[str]) -> None:
 async def _add_platformio_options(pio_options):
     # Add includes at the very end, so that they override everything
     for key, val in pio_options.items():
-        if key == "build_flags" and not isinstance(val, list):
+        if key in ["build_flags", "lib_ignore"] and not isinstance(val, list):
             val = [val]
         cg.add_platformio_option(key, val)
 

